TDDPracticeTests/TDDPracticeTests/CalculateNumber/CalculatorTests.cs

```diff
@@ -0,0 +1,75 @@
+﻿using FluentAssertions;
+using Xunit.Abstractions;
+
+namespace TestCasesProject.CalculatorCaseStudy
+{
+
+    public class CalculatorTests : IDisposable
+    {
+        //dispose all running instances
+        public void Dispose()
+        {
+        }
+
+        [Theory]
+        [MemberData(nameof(TestData))]
+        public void Return_Sum_Of_Numbers(string numbers, int expectedResult)
+        {
+            //Act
+            int result = Calculator.Add(numbers);
+
+            //Assert
+            result.Should().NotBe(0);
+            result.Should().BePositive();
+            result.Should().Be(expectedResult);
+        }
+
+        public static IEnumerable<object[]> TestData => new[]
+        {
+            new object[]{ "1,2", 3 },
+            new object[]{ "7,2", 9 },
+            new object[]{ "1,2,5", 8 },
+            new object[]{ "7,2,9,2", 20 },
+            new object[]{ "7\n,2,9,2", 20 },
+            new object[]{ "//;\n1;2", 3 },
+            new object[]{"//~;\n1~;2~;3", 6}
+        };
+
+        [Fact]
+        public void Return_Exception_If_Numbers_Have_Comma_With_New_Line()
+        {
+            //arrange
+            string numbers = "1,\n2";
+
+            //Act
+            Action action = () => Calculator.Add(numbers);
+
+            //Assert
+            action.Should().ThrowExactly<ArgumentException>().WithMessage("Wrong input.");
+        }
+
+        [Fact]
+        public void Return_Exception_If_Numbers_Contains_Negative_Number()
```

Comment:
please follow best practices for naming tests https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices

Reply:
Thanks for sharing this article. Please check the new committed code and let me know, if any other changes needed.